cloud/ec2.go

```diff
@@ -907,14 +917,37 @@ func (m *ec2Manager) GetDNSName(ctx context.Context, h *host.Host) (string, erro
 	h.StartTime = *instance.LaunchTime
 	h.VolumeTotalSize, err = getVolumeSize(ctx, m.client, h)
 	if err != nil {
-		return "", errors.Wrapf(err, "error getting volume size for host %s", h.Id)
+		return nil, errors.Wrapf(err, "error getting volume size for host %s", h.Id)
 	}
 	if err := h.CacheHostData(); err != nil {
-		return "", errors.Wrap(err, "error updating host document in db")
+		return nil, errors.Wrap(err, "error updating host document in db")
+	}
+
+	return instance, nil
+}
+
+// GetDNSName returns the DNS name for the host.
+func (m *ec2Manager) GetDNSName(ctx context.Context, h *host.Host) (string, error) {
+	instance, err := m.retrieveInstance(ctx, h)
+	if err != nil {
+		return "", err
+	}
+
+	// set IPv6 address, if applicable
+	for _, networkInterface := range instance.NetworkInterfaces {
```

Comment:
In `spawnOnDemandHost` and `spawnSpotHost` you set only the 0th network interface. Only check the 0th for consistency here?

Reply:
We talked about potentially there being two network interfaces bc Amazon would add one, so I wanted to handle the case where that interface is first. This seems immediately answerable by testing via commandline when we have that option.